src/middlewared/middlewared/plugins/filesystem_/acl_base.py

```diff
@@ -292,11 +292,21 @@ def setacl(self, job, data):
     @accepts(
         Str('path'),
         Bool('simplified', default=True),
+        Bool('resolve_ids', default=False),
     )
-    def getacl(self, path, simplified):
+    def getacl(self, path, simplified, resolve_ids):
```

Comment:
Do we need to make this method a job since, historically, resolving uid/gids to their respective names can take quite awhile?

Reply:
Changing to a job is a hugely intrusive change IMHO (this is used lots of places), but there are pathological edge cases in some environments. Perhaps a better option is to specify timeout in the `id_to_name()` method and return None if it takes something like more than 5 seconds to complete the request. In theory this should be enough to get winbind to cache response.

Reply:
It's definitely intrusive 👀 . I agree, a timeout value would be good.